src/override/rust.cc

```diff
@@ -0,0 +1,37 @@
+#include "../mem/slowalloc.h"
+#include "../snmalloc.h"
+
+#include <cstring>
+
+#ifndef SNMALLOC_EXPORT
+#  define SNMALLOC_EXPORT
+#endif
+
+using namespace snmalloc;
+
+extern "C" SNMALLOC_EXPORT void* rust_alloc(size_t alignment, size_t size)
+{
+  return ThreadAlloc::get_noncachable()->alloc(aligned_size(alignment, size));
+}
+
+extern "C" SNMALLOC_EXPORT void
+rust_dealloc(void* ptr, size_t alignment, size_t size)
+{
+  ThreadAlloc::get_noncachable()->dealloc(ptr, aligned_size(alignment, size));
+}
+
+extern "C" SNMALLOC_EXPORT void*
+rust_realloc(void* ptr, size_t alignment, size_t old_size, size_t new_size)
+{
+  size_t aligned_old_size = aligned_size(alignment, old_size),
+         aligned_new_size = aligned_size(alignment, new_size);
+  if (aligned_old_size == aligned_new_size)
```

Comment:
As `aligned_size` does not return a size that is necessarily a sizeclass, this check is too conservative now.

It should be 
```C++
if (size_to_sizeclass(aligned_old_size) == size_to_sizeclass(aligned_new_size))
```